docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+improving usability and consistency across different resources in CAPI and with the rest of the ecosystem.
+
+# Motivation
+
+The Cluster API community recognize that nowadays Cluster API and Kubernetes users are rightfully focused on
+building higher systems and great applications on top on those platforms, which is great.
+
+However, as a consequence of this shifted focus, most of the users don’t have time to become deep expert of Cluster API
+like the first wave of adopters, and also Cluster API maintainers would like they don’t have to.
+
+The effect of the trend above is the blurring of the lines not only between different Cluster API components, but also
+between Cluster API, core Kubernetes and a few other broadly adopted tools like Helm or Flux (and to some extents,
+also with many others awesome tools in the ecosystem).
+
+This is why Cluster API status must become simpler to understand for users, and also more consistent not only across
+different CAPI resources, but with Kubernetes core and ideally with the entire ecosystem.
+
+### Goals
+
+- Review and standardize the usage of the concept of readiness across Cluster API resources.
+    - Drop or amend improper usage of readiness
+    - Make the concept of Machine readiness extensible, thus allowing providers or external system to inject their readiness checks.
+- Review and standardize the usage of the concept of availability across Cluster API resources.
+    - Make the concept of Cluster Availability extensible, thus allowing providers or external system to inject their availability checks.
+- Bubble up more information about both CP and worker Machines, ensuring consistent way across Cluster API resources.
+    - Standardize replica counters and bubble them up to the Cluster resource.
+    - Standardize control plane, MachineDeployment, Machine pool availability, and bubble them up to the Cluster resource.
+- Introduce missing signals about connectivity to workload clusters, thus enabling to mark all the conditions
+  depending on such connectivity being working with status Unknown after a certain amount of time.
+- Introduce a cleaner signal about Cluster API resources lifecycle transitions, e.g. scaling up or updating.
+- Ensure everything in status can be used as a signal informing monitoring tools/automation on top of Cluster 
+  about lifecycle transitions/state of the Cluster and the underlying components as well.
+
+### Non-Goals/Future Work
+
+- Resolving all the idiosyncrasies that exists in Cluster API, core Kubernetes, the rest of the ecosystem.
+  (Let’s stay focused in Cluster API and keep improving incrementally).
+
+## Proposal
+
+This proposal groups a set of changes to status fields in Cluster API resources.
+
+Some of those changes could be considered straight forward, e.g.
+
+- K8s API conventions suggest to deprecate and remove `phase` fields from status, Cluster API is going to align to this recommendation
```

Comment:
One positive note about `phase` is the immediate feedback that it provides when running things like `kubectl get machines` or other resources. It provides a clear and simple feedback on "where are we at".

Given that this change is geared towards end users, do we agree that we need a replacement first before we deprecate something that's considered generally useful?

Reply:
I can see both reasons.

[K8s guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) have a pretty good argument about dropping phase, and unless I'm mistaken, they have already dropped it from print columns in most of the core types.

But also I get it is intuitive for users.

What made me lean towards the decision to drop, is the fact that IMO phases in CAPI objects are mostly about the provisioning workflow, while they are almost useless for the rest of the lifecycle of the objects (except might be for deletion).

But happy to chat about it

Reply:
Considering also other discussions, we are keeping phases (and intentionally diverging from guidelines for this point). Also added a paragraph explaining why we took this decision